src/main/java/codeu/controller/UserEditServlet.java

```diff
@@ -0,0 +1,81 @@
+package codeu.controller;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import codeu.model.store.basic.UserStore;
+import codeu.model.data.User;
+
+public class UserEditServlet extends HttpServlet {
+
+  /**
+   * Store class that gives access to Users.
+   */
+  private UserStore userStore;
+
+  /**
+   * Set up state for handling about page-related requests. This method is only
+   * called when running in a server, not when running in a test.
+   */
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(UserStore.getInstance());
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common
+   * setup method for use by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+    /**
+   * This function fires when a user requests the /users URL. It simply forwards
+   * the request to users.jsp.
+   */
+    @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+    request.getRequestDispatcher("/WEB-INF/view/useredit.jsp").forward(request, response);
+  }
+
+  /**
+   * This function handles post requests to update the User with all information sent. It creates a
+   * user and updates the userstore with the new user information.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+    String username = request.getParameter("user");
```

Comment:
This should be String username = (String) request.getSession().getAttribute("user");

Reply:
Oops, this should be as you suggested, I thought you were pointing to doGet, mb.